### PR TITLE
Fixed predictive back gesture not working on OnePlus devices

### PR DESCRIPTION
**Pull request by arkivanov**
On some OnePlus devices with Android 14 the predictive back gesture is not working if the app depends on a new version of androidx.activity. The root cause is that those Android versions dispatch onBackStarted when the gesture finger is released. The new versions of OnBackPressedDispatcher now use NavigationEventDispatcher and dispatch onBackCancelled when onBackStarted is received while the back gesture is already in progress.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved predictive back gesture handling so cancelled, restarted, or completed gestures behave more reliably during navigation transitions.
  * Fixed edge cases where back progress could be lost or stack animations could finish/cancel inconsistently.
* **Tests**
  * Updated predictive back gesture tests to better simulate long-running finish/cancel flows.
  * Added new coverage for “progress → cancel → back” and “progress → cancel → restart” scenarios to prevent regressions.

### CODE REVIEW

File: extensions-compose-experimental/src/commonMain/kotlin/com/arkivanov/decompose/extensions/compose/experimental/stack/animation/DefaultStackAnimation.kt [hunk 1]
```diff
@@ -240,23 +242,31 @@ internal class DefaultStackAnimation<C : Any, T : Any>(
...
-                state = State.Started(backEvent)
+            val currentState = state
+            if (currentState is State.Idle) {
+                val childScope = parentScope + Job(parentScope.coroutineContext[Job])
+                state = State.Started(backEvent, childScope)
+            } else if (currentState is State.Cancelling) {
+                currentState.scope.cancel()
+                val newScope = parentScope + Job(parentScope.coroutineContext[Job])
+                state = State.Started(currentState.lastBackEvent, newScope)
+                onBackProgressed(currentState.lastBackEvent)
```
[the remark this answers is not in the document]
Comment (arkivanov) [reply extensions-compose-experimental/src/commonMain/kotlin/com/arkivanov/decompose/extensions/compose/experimental/stack/animation/DefaultStackAnimation.kt:260]: This is intentional in this case. If the gesture is restarted while cancellation is in progress, we want to continue from the previous progress.
